docs/structure-of-a-contract.rst

```diff
@@ -42,6 +42,43 @@ Functions are the executable units of code within a contract.
 and have different levels of visibility (:ref:`visibility-and-getters`)
 towards other contracts. Functions must be decorated with either @public or @private.
 
+Default function
+----------------
+
+A contract can also have a default function, which is executed on a call to the contract if no other functions match the given function identifier (or if none was supplied at all, such as through someone sending it Eth). It is the same construct as fallback functions `in Solidity <https://solidity.readthedocs.io/en/latest/contracts.html?highlight=fallback#fallback-function>`_. 
+
+This function is always named `__default__` and must be annotated with `@public`. It cannot have arguments and cannot return anything.
+
+If the function is annotated as `@payable`, this function is executed whenever the contract is sent Ether (without data). This is why the default function cannot accept arguments and return values - it is a design decision of Ethereum to make no differentiation between sending ether to a contract or a user address.
+
+**Example:**
+
+::
+
+    Payment: event({amount: int128, from: indexed(address)})
+    
+    @public
+    @payable
+    def __default__():
+        log.Payment(msg.value, msg.sender)
+
+
+Considerations
+~~~~~~~~~~~~~~
+
+Unlike Solidity, Vyper generates a default function is one isn't found, in the form of a REVERT call. Note that this still `generates an exception <https://github.com/ethereum/wiki/wiki/Subtleties>`_ and thus will not succeed in receiving funds. 
+
+Ethereum specifies that the operations will be rolled back if the contract runs out of gas in execution. ``send`` calls to the contract come with a free stipend of 2300 gas, which does not leave much room to perform other operations except basic logging. **However**, if the sender includes a higher gas amount through a ``call`` instead of ``send``, then more complex functionality can be run.
```

Comment:
This is more nuanced than your description. I can call a default/fallback directly from a raw transaction (in a contract or from an EOA) and the gas stipend is more than 2300. This stipend only holds when when an `address.transfer(uint)` is called, as implemented by Solidity and Vyper compilers as a mitigation against excessive gas usage.

It is therefore considered a *best practice* to ensure your payable default function is compatible with the 2300 gas stipend. If it is currently an error, then overusing the gas stipend should be a warning instead. There are legitimate use cases for a default that exceeds this amount. Lots of nuance here.

@jacqueswww, do you agree?